Common platform

alizelzele opened the pull request:
changing the common mobile platform to support both platforms with only one class make it easier to maintenance + normally the functions that are required for automation are common and dont need any specific driver commands.

alizelzele commented:
@mach6 the first 3 commits are the pull requests i made. can you check the last commit and give me your opinion on this change?

mach6 commented:
@alizelzele I took a look at the last commit in this PR. I have concerns about the approach/changes but don't want to abandon the idea. Let's please focus on getting your other PR's in before proceeding with the "common platform" changes proposed here. Afterwards, we will circle back.

looping in @sebady

mach6 commented:
@alizelzele @sebady  due to the _size_ of the "common platform" changes, does it make sense to work them on common branch?

sebady commented:
I think we need to segregate the "common platform" commit from the others in this PR.

mach6 commented:
@alizelzele can you please follow-up on this now?

mach6 commented:
@alizelzele bumping this.. Do you still see these changes as something you want to see merged? if so, will you please update the PR? I'd like to get this closed out.

alizelzele commented:
@mach6 I completely forgot about this one. will work on it again, and update the code

mach6 commented:
@alizelzele Will you please look into the CI failure? It looks like the archetype build is broken due to this change set.

mach6 commented on client/src/main/java/com/paypal/selion/platform/grid/Grid.java:
```diff
@@ -112,7 +115,20 @@ public static RemoteWebDriver driver() {
+    public static SeLionAppiumIOSDriver iOSDriver(){
+    	return SeLionAppiumIOSDriver.class.cast(driver());
+    }
+    
```
These three methods are appium only. I'd like Grid() to remain neutral w.r.t to web/mobile driver type

mach6 commented:
Please update copyright for changes files to include the year 2016

alizelzele commented:
@mach6 Fixed all the issues from checkers, updated copyrights, and move appium driver methods out of Grid. As discussed earlier, this will remove 2 different platform and try to make both of them work as one. That was the reason archytype generator was failing.
It still needs some Updates to remove the old ones completely out of it

alizelzele committed:
only one platform for all mobile platforms

1. using only one mobile platform and remove other not required elements
2. extending Abstract Element as base Element to support already defined functionality for Web
3. adding wait for multiple elements in WaitDriver Utils
4. updating tests with the new system
5. updating code generator for new system and removing suport for android and ios directly
6. updating archetype to generate current archetypes based on new changes.

alizelzele commented:
@mach6 I am closing this one since it is super trouble to rebase it to latest. I'll try my best to make separate smaller pull requests and not to touch core components (as much as possible) :-)
